src/backends/r1cs/mod.rs

```diff
@@ -463,6 +467,21 @@ where
             witness[var.index] = val;
         }
 
+        // print out the log info
+        for (_, span, var_info) in &self.log_info {
```

Comment:
what about that msg?

Reply:
currently the native code doesn't support string type, so the msg for each log can't be customized.
I guess this will be only possible after it supported string type or string stdlib.